Scripts/Tracking/CharacterController/CharacterControllerMover.cs

```diff
@@ -0,0 +1,148 @@
+﻿namespace VRTK.Core.Tracking.CharacterController
+{
+    using System;
+    using UnityEngine;
+    using UnityEngine.Events;
+    using VRTK.Core.Data.Attribute;
+    using VRTK.Core.Data.Source;
+    using VRTK.Core.Process;
+
+    /// <summary>
+    /// Moves a <see cref="CharacterController"/> by following the movement of two <see cref="GameObject"/>s.
+    /// </summary>
+    public class CharacterControllerMover : MonoBehaviour, IProcessable
+    {
+        /// <summary>
+        /// The collision happening during a move of a <see cref="CharacterController"/>.
+        /// </summary>
+        [Flags]
+        public enum CollisionFlags
+        {
+            /// <summary>
+            /// No collision occured.
+            /// </summary>
+            None = 1 << 0,
+            /// <summary>
+            /// A collision with the sides occured.
+            /// </summary>
+            Sides = 1 << 1,
+            /// <summary>
+            /// A collision with the top occured.
+            /// </summary>
+            Above = 1 << 2,
+            /// <summary>
+            /// A collision with the bottom occured.
+            /// </summary>
+            Below = 1 << 3
+        }
+
+        /// <summary>
+        /// Defines the event with the <see cref="CollisionFlags"/>.
+        /// </summary>
+        [Serializable]
+        public class UnityEvent : UnityEvent<CollisionFlags>
```

Comment:
Class should appear above the enum according to the updated contributing guidelines